### ISSUE TITLE
Can't arm/disarm ResultCode -4106

### ISSUE THREAD

**Issue by austinmroczek** [craigjmidwinter/total-connect-client#85]
First reported at https://github.com/home-assistant/core/issues/37323 by @SirTerrific

```txt

2020-07-01 23:39:45 ERROR (SyncWorker_18) [root] Could not disarm system. ResultCode: -4106. ResultData: Invalid user code. Please try again.
2020-07-01 23:39:45 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection.140117669686864] TotalConnect failed to disarm MONTREAL.
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 130, in handle_call_service
    connection.context(msg),
  File "/usr/src/homeassistant/homeassistant/core.py", line 1260, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1295, in _execute_service
    await handler.func(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 209, in handle_service
    self._platforms.values(), func, call, required_features
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 454, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 595, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 485, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/alarm_control_panel/__init__.py", line 128, in async_alarm_disarm
    await self.hass.async_add_executor_job(self.alarm_disarm, code)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/totalconnect/alarm_control_panel.py", line 119, in alarm_disarm
    raise HomeAssistantError(f"TotalConnect failed to disarm {self._name}.")
homeassistant.exceptions.HomeAssistantError: TotalConnect failed to disarm MONTREAL.
2020-07-02 00:06:15 WARNING (MainThread) [homeassistant.helpers.entity] Update of alarm_control_panel.montreal is taking over 10 seconds
2020-07-02 00:06:46 WARNING (MainThread) [homeassistant.helpers.entity] Update of alarm_control_panel.montreal is taking over 10 seconds
2020-07-02 00:10:23 WARNING (MainThread) [homeassistant.helpers.entity] Update of alarm_control_panel.montreal is taking over 10 seconds

```

**Comment by austinmroczek** [comment]
@sirterrific a few questions for you

a) You said that TotalConnect never worked for you.  Have you used Home Assistant for a long time and TotalConnect never worked, or did you just started using it?

b) What alarm panel hardware are you using?  

c) Please try the Troubleshooting directions (at https://github.com/craigjmidwinter/total-connect-client) and pasting the result here.  Recommend trying twice, once with your real usercode, and once with with "-1".

**Comment by SirTerrific** [comment]
a) I misexpressed myself. This is my first installation and first configuration with Total Connect.

b) A Lynx 5210

c) 
`Auto Bypass Low Battery: False
Valid Credentials: True
Module Flags:
  Security: 1
  Video: 1
  Automation: 1
  GPS: 1
  VideoPIR: 1
  ReadState: 1
  IsAlexaEnabled: 0
Username: 
UserID: 
Master User: False
User Administrator: False
Configuration Administrator: False
User features:
  Master: 0
  User Administration: 0
  Configuration Administration: 0

LOCATIONS: 1

LOCATION 788957 - MONTREAL

PhotoURL: None
SecurityDeviceID: 
DeviceList: OrderedDict([('DeviceInfoBasic', [OrderedDict([('DeviceID', ), ('DeviceName', 'Security Panel'), ('DeviceClassID', 1), ('DeviceSerialNumber', ''), ('DeviceFlags', 'PromptForUserCode=0,PromptForInstallerCode=0,PromptForImportSecuritySettings=0,AllowUserSlotEditing=0,CalCapable=1,CanBeSentToPanel=0,CanArmNightStay=0,CanSupportMultiPartition=0,PartitionCount=0,MaxPartitionCount=0,OnBoardingSupport=0,PartitionAdded=0,DuplicateUserSyncStatus=0,PanelType=8,PanelVariant=1,BLEDisarmCapable=0,ArmHomeSupported=0,DuplicateUserCodeCheck=1,CanSupportRapid=0,IsKeypadSupported=1,WifiEnrollmentSupported=1,IsConnectedPanel=0,ArmNightInSceneSupported=0,BuiltInCameraSettingsSupported=0,ZWaveThermostatScheduleDisabled=0'), ('SecurityPanelTypeID', None), ('DeviceSerialText', None)])])])
AcLoss: False
LowBattery: False
IsCoverTampered: False
Arming State: 10203
LocationModuleFlags:
  Security: 1
  Video: 0
  Automation: 0
  GPS: 0
  VideoPIR: 0
  TimeTriggeredEvent: 0
  TemperatureUnits: C
  ConfigureContent: 0
  SyncLocation: 0
  ConfigureSlideshow: 0
  TimezoneOffset: -4.0
  SmartAction: 0
  CustomArm: 0
  NoTriggerScene: 0
  NoScene: 0
  AutoSyncEnabled: 1
  WiFIThermostatEnabled: 1
  SupportsOnlyHDPhotos: 0
  SyncStatusSupported: 1
  WiFiHBSupported: 1
  DoorBellSupported: 0
  masterUserCodeSync: 0
  GeofenceStatus: 0
  RSISupported: 0
  VideoServiceEnabled: 0
  HasAddressUpdated: 1
  WifiGaragedoorSupported: 0
  OutboundServiceEnabled: 0
  HasSmartScenes: 0
  MotionViewerServiceEnabled: 0
  VavEnabled: 0

ZONES: 12

Zone 2 - Avant Porte
Partition: 1    Type: 0 Status: 0

Zone 3 - Arrière Porte
Partition: 1    Type: 0 Status: 0

Zone 4 - Dét. mouvement
Partition: 1    Type: 0 Status: 64

Zone 5 - Dét. de fumée
Partition: 1    Type: 9 Status: 0

Zone 95 - Feu
Partition: 1    Type: 9 Status: 0

Zone 99 - Police
Partition: 1    Type: 0 Status: 0

Zone 180 - Température
Partition: 1    Type: 0 Status: 0

Zone 181 - Température
Partition: 1    Type: 0 Status: 0

Zone 182 - Température
Partition: 1    Type: 0 Status: 0

Zone 183 - Température
Partition: 1    Type: 0 Status: 0

Zone 184 - Température
Partition: 1    Type: 0 Status: 0

Zone 185 - Température
Partition: 1    Type: 0 Status: 0`

**Comment by austinmroczek** [comment]
@SirTerrific I fixed a few things in the code.  Please try this:

`wget https://raw.githubusercontent.com/craigjmidwinter/total-connect-client/master/total_connect_client/TotalConnectClient.py` (same as before, but new code)
`wget https://raw.githubusercontent.com/craigjmidwinter/total-connect-client/master/total_connect_client/live_test_disarm.py`
`wget https://raw.githubusercontent.com/craigjmidwinter/total-connect-client/master/total_connect_client/live_test_arm.py`

`python3 live_test_arm.py username password usercode`
`python3 live_test_disarm.py username password usercode`

You should get a simple message like `Arm success.` and your system will Arm Stay.  Same for Disarm.

**Comment by SirTerrific** [comment]
HI, it's working now

**Comment by austinmroczek** [comment]
This is fixed by PR #88.  Version 0.55.5 was uploaded to pypi.  Now need to update Home Assistant.

**Comment by austinmroczek** [comment]
In the process of updating the code for Home Assistant, I've realized a few things that need to be documented.  

The usercode is specific to the combination of the user AND the device (which is tied to the location).  

This means we've been handling usercodes wrong for a while, where we assumed the usercode is tied to the user.  I didn't realize this until I started to implement ValidateUserCode https://rs.alarmnet.com/TC21api/tc2.asmx?op=ValidateUserCode so we could add to the configuration flow in Home Assistant.  We probably never ran into this issue because the vast majority of users only have one location.

We need to keep a usercode for each location, not for the client/user.
